src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1219,215 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
+                        {
+                            jobSchedules.Add(AgentUtilities.ConvertToAgentScheduleInfo(schedule));
+                        }
+                        result.Schedules = jobSchedules.ToArray();
+                        // Add histories
+                        int count = dt.Rows.Count;
+                        List<AgentNotebookHistoryInfo> notebookHistories = new List<AgentNotebookHistoryInfo>();
+                        if (count > 0)
+                        {
+                            var job = dt.Rows[0];
+                            Guid jobId = (Guid)job[AgentUtilities.UrnJobId];
+                            int runStatus = Convert.ToInt32(job[AgentUtilities.UrnRunStatus], System.Globalization.CultureInfo.InvariantCulture);
+                            var t = new LogSourceJobHistory(parameters.JobName, sqlConnInfo, null, runStatus, jobId, null);
+                            var tlog = t as ILogSource;
+                            tlog.Initialize();
+                            var logEntries = t.LogEntries;
+                            var jobHistories = AgentUtilities.ConvertToAgentNotebookHistoryInfo(logEntries, job, steps);
+                            Dictionary<string, AgentNotebookHistoryInfo> allNotebookHistoriesHashTable = new Dictionary<string, AgentNotebookHistoryInfo>();
+                            // Finally add the job histories
+                            DataTable materializedNotebookTable = NotebookHistoryDataset.Tables[0];
+                            foreach (DataRow materializedNotebookRow in materializedNotebookTable.Rows){
+                                string materializedRunDateTime = materializedNotebookRow["run_date"].ToString() + materializedNotebookRow["run_time"].ToString();
+                                AgentNotebookHistoryInfo notebookHistory = new AgentNotebookHistoryInfo();
+                                notebookHistory.MaterializedNotebookId = (int)materializedNotebookRow["materialized_id"];
+                                allNotebookHistoriesHashTable.Add(materializedRunDateTime, notebookHistory);
+                            }
+                            foreach (var jobHistory in jobHistories){
+                                string jobRuntime = jobHistory.RunDate.ToString("yyyyMMddHHmmss");
+                                AgentNotebookHistoryInfo notebookHistory = jobHistory;
+                                if(allNotebookHistoriesHashTable.ContainsKey(jobRuntime))
+                                    notebookHistory.MaterializedNotebookId = allNotebookHistoriesHashTable[jobRuntime].MaterializedNotebookId;
+                                notebookHistories.Add(notebookHistory);
+                            }
+                            result.Histories = notebookHistories.ToArray();
+                            result.Success = true;
+                            tlog.CloseReader();
+                            await requestContext.SendResult(result);
```

Comment:
This should be in a finally block with the try starting right above creating the LogSourceJobHistory (or whever the reader is opened)